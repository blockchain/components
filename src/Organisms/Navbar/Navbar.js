import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: ${(props) => props.height};
  padding: 0 2rem;
  box-sizing: border-box;

  @media (min-width: 48rem) {
    padding: 0 3rem;
  }
`

const Navbar = ({ children, height, ...rest }) => (
  <Wrapper height={height} {...rest}>
    {children}
  </Wrapper>
)

Navbar.propTypes = {
  children: PropTypes.node,
  height: PropTypes.string.isRequired,
}

Navbar.defaultProps = {
  height: '60px',
}

export default Navbar
