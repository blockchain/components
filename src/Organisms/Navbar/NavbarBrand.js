import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: relative;
`

const NavbarBrand = ({ children, ...rest }) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

NavbarBrand.propTypes = {
  children: PropTypes.node,
  height: PropTypes.string.isRequired,
}

NavbarBrand.defaultProps = {
  height: '60px',
}

export default NavbarBrand
