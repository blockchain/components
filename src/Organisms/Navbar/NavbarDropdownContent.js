// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

const Wrapper = styled.div`
  width: 100%;
  z-index: 10;

  @media (min-width: ${(props) => props.theme.screenSizes.sm}) {
    position: absolute;
    width: auto;
    margin: 0.25rem 0;
    background-color: transparent;
  }
`

const Container = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;

  & > :last-child {
    margin-bottom: 0 !important;
  }

  @media (max-width: ${(props) => props.theme.screenSizes.sm}) {
    padding: 1.1rem 0;
  }

  @media (min-width: ${(props) => props.theme.screenSizes.sm}) {
    margin-top: 15px;
    padding: 1.5rem;
    border-radius: 0.25rem;
    box-sizing: border-box;
    background-color: ${(props) =>
      props.theme['navbarDropdownContentBackgroundColor']};
    box-shadow: 0 10px 60px -10px ${(props) => props.theme['navbarDropdownContentShadowColor']},
      0 8px 16px ${(props) => props.theme['navbarDropdownContentShadowColor2']};

    &::after,
    &::before {
      bottom: 100%;
      left: 10%;
      border: solid transparent;
      content: ' ';
      height: 0;
      width: 0;
      position: absolute;
      pointer-events: none;
    }

    &::after {
      border-color: transparent;
      border-bottom-color: ${(props) =>
        props.theme['navbarDropdownContentBackgroundColor']};
      border-width: 10px;
      margin-left: -10px;
    }

    &::before {
      border-color: transparent;
      border-bottom-color: ${(props) =>
        props.theme['navbarDropdownContentBackgroundColor']};
      border-width: 12px;
      margin-left: -12px;
    }
  }
`

type PropsType = {
  +children?: React.Node,
}

const NavbarDropdownContent = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>
    <Container>{children}</Container>
  </Wrapper>
)

NavbarDropdownContent.propTypes = {
  children: PropTypes.node,
}

export default NavbarDropdownContent
