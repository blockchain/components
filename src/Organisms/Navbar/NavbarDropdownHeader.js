import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.span`
  font-family: ${(props) => props.theme['fontPrimary']};
  font-size: 1rem;
  font-weight: 600;
  color: ${(props) => props.theme['navbarDropdownHeaderColor']};
  cursor: pointer;
  display: none;

  @media (min-width: 48rem) {
    display: block;
  }
`

const NavbarDropdownHeader = ({ children, ...rest }) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

NavbarDropdownHeader.propTypes = {
  children: PropTypes.node,
}

export default NavbarDropdownHeader
