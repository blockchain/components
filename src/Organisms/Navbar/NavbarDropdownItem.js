// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

import Link from '../../Atoms/Links/Link'

type PropsType = {
  +children?: React.Node,
}

const Wrapper = styled(Link)`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
  flex-wrap: nowrap;
  height: 2.5rem;
  margin-bottom: 1.5rem;
  white-space: nowrap;
  text-decoration: none;

  @media (min-width: ${(props) => props.theme.screenSizes.sm}) {
    &:hover span:first-child {
      text-decoration: underline;
    }
  }
`

const NavbarDropdownItem = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

NavbarDropdownItem.propTypes = {
  children: PropTypes.node,
}

export default NavbarDropdownItem
