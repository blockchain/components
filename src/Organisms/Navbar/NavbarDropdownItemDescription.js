// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

import Text from '../../Atoms/Typography/Text'
import { screenSize, theme } from '../../Tools/interpolation'

export type PropsType = {
  +children?: React.Node,
}

const Wrapper = styled(Text)`
  color: ${theme('navbarDropdownDescriptionMobileColor')};

  @media (min-width: ${screenSize('sm')}) {
    color: ${theme('navbarDropdownDescriptionColor')};
  }
`

const NavbarDropdownItemDescription = ({ children, ...rest }: PropsType) => (
  <Wrapper size="0.875rem" weight={500} {...rest}>
    {children}
  </Wrapper>
)

NavbarDropdownItemDescription.propTypes = {
  children: PropTypes.node,
}

export default NavbarDropdownItemDescription
