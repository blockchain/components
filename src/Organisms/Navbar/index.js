import Navbar from './Navbar'
import NavbarBrand from './NavbarBrand'
import NavbarDropdownContent from './NavbarDropdownContent'
import NavbarDropdownHeader from './NavbarDropdownHeader'
import NavbarHeader from './NavbarHeader'
import NavbarMenu from './NavbarMenu'
import NavbarNav from './NavbarNav'
import NavbarNavDropdown from './NavbarNavDropdown'
import NavbarNavItem from './NavbarNavItem'
import NavbarToggler from './NavbarToggler'

export default {
  Navbar,
  NavbarBrand,
  NavbarDropdownContent,
  NavbarDropdownHeader,
  NavbarHeader,
  NavbarMenu,
  NavbarNav,
  NavbarNavDropdown,
  NavbarNavItem,
  NavbarToggler,
}
