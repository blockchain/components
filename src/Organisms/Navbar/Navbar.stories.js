import React from 'react'
import styled from 'styled-components'
import { storiesOf } from '@storybook/react'
import Button from '../../Atoms/Buttons/Button'
import Image from '../../Atoms/Images/Image'
import Section from '../../Atoms/Containers/Section'
import Navbar from './Navbar'
import NavbarBrand from './NavbarBrand'
import NavbarDropdownContent from './NavbarDropdownContent'
import NavbarDropdownHeader from './NavbarDropdownHeader'
import NavbarDropdownItem from './NavbarDropdownItem'
import NavbarDropdownItemDescription from './NavbarDropdownItemDescription'
import NavbarDropdownItemTitle from './NavbarDropdownItemTitle'
import NavbarHeader from './NavbarHeader'
import NavbarMenu from './NavbarMenu'
import NavbarNav from './NavbarNav'
import NavbarNavDropdown from './NavbarNavDropdown'
import NavbarNavItem from './NavbarNavItem'
import NavbarToggler from './NavbarToggler'

const OrientSection = styled(Section)`
  background-color: ${(props) => props.theme.palette['orient']};
`
const Wrapper = (props) => <OrientSection {...props} />

const ButtonLogin = () => <Button>Login</Button>

const Content = () => (
  <React.Fragment>
    <NavbarDropdownItem>
      <NavbarDropdownItemTitle>Link 1</NavbarDropdownItemTitle>
      <NavbarDropdownItemDescription>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </NavbarDropdownItemDescription>
    </NavbarDropdownItem>
    <NavbarDropdownItem>
      <NavbarDropdownItemTitle>Link 2</NavbarDropdownItemTitle>
      <NavbarDropdownItemDescription>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </NavbarDropdownItemDescription>
    </NavbarDropdownItem>
    <NavbarDropdownItem>
      <NavbarDropdownItemTitle>Link 3</NavbarDropdownItemTitle>
      <NavbarDropdownItemDescription>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </NavbarDropdownItemDescription>
    </NavbarDropdownItem>
  </React.Fragment>
)

storiesOf('Organisms/Navbar', module)
  .add('Example 1', () => (
    <Wrapper>
      <Navbar>
        <NavbarHeader>
          <NavbarBrand>
            <Image name="brand" width="175px" />
          </NavbarBrand>
        </NavbarHeader>
        <NavbarMenu>
          <NavbarNav width="auto">
            <NavbarNavDropdown>
              <NavbarDropdownHeader>Header 1</NavbarDropdownHeader>
              <NavbarDropdownContent>
                <Content />
              </NavbarDropdownContent>
            </NavbarNavDropdown>
            <NavbarNavDropdown>
              <NavbarDropdownHeader>Header 2</NavbarDropdownHeader>
              <NavbarDropdownContent>
                <Content />
              </NavbarDropdownContent>
            </NavbarNavDropdown>
          </NavbarNav>
          <NavbarNav width="auto">
            <NavbarNavItem>
              <ButtonLogin />
            </NavbarNavItem>
          </NavbarNav>
        </NavbarMenu>
        <NavbarToggler />
      </Navbar>
    </Wrapper>
  ))
  .add('Example 2', () => (
    <Wrapper>
      <Navbar>
        <NavbarHeader>
          <NavbarBrand>
            <Image name="brand" width="175px" />
          </NavbarBrand>
        </NavbarHeader>
        <NavbarMenu toggled>
          <NavbarNav width="auto">
            <NavbarNavDropdown>
              <NavbarDropdownHeader>Header 1</NavbarDropdownHeader>
              <NavbarDropdownContent>
                <Content />
              </NavbarDropdownContent>
            </NavbarNavDropdown>
            <NavbarNavDropdown>
              <NavbarDropdownHeader>Header 2</NavbarDropdownHeader>
              <NavbarDropdownContent>
                <Content />
              </NavbarDropdownContent>
            </NavbarNavDropdown>
          </NavbarNav>
          <NavbarNav width="auto">
            <NavbarNavItem>
              <ButtonLogin />
            </NavbarNavItem>
          </NavbarNav>
        </NavbarMenu>
        <NavbarToggler />
      </Navbar>
    </Wrapper>
  ))
