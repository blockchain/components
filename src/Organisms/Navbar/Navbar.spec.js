import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import Navbar from './Navbar'

describe('Navbar', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Navbar />)

    expect(tree.find(Navbar)).toMatchSnapshot()
  })
})
