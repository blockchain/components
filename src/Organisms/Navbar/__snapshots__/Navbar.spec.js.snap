// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 60px;
  padding: 0 2rem;
  box-sizing: border-box;
}

@media (min-width:48rem) {
  .c0 {
    padding: 0 3rem;
  }
}

<Navbar
  height="60px"
>
  <styled.div
    height="60px"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: ",
              [Function],
              ";
  padding: 0 2rem;
  box-sizing: border-box;

  @media (min-width: ",
              [Function],
              ") {
    padding: 0 3rem;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      height="60px"
    >
      <div
        className="c0"
        height="60px"
      />
    </StyledComponent>
  </styled.div>
</Navbar>
`;

exports[`Navbar should render correctly with height 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 60px;
  padding: 0 2rem;
  box-sizing: border-box;
}

@media (min-width:48rem) {
  .c0 {
    padding: 0 3rem;
  }
}

<Navbar
  height="60px"
>
  <styled.div
    height="60px"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: ",
              [Function],
              ";
  padding: 0 2rem;
  box-sizing: border-box;

  @media (min-width: ",
              [Function],
              ") {
    padding: 0 3rem;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      height="60px"
    >
      <div
        className="c0"
        height="60px"
      />
    </StyledComponent>
  </styled.div>
</Navbar>
`;
